protectedts: delete deprecated code

dt opened the pull request.

dt committed:
protectedts: remove UseMetaTable knob

Release note: none.
Epic: none.

dt committed:
kv/kvserver/protectedts: remove useDeprecatedProtectedTSStorage

Release note: none.
Epic: none.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/153368)
<!-- Reviewable:end -->

dt commented:
2¢: `hide whitespace` is useful on this one.

dt committed:
pkg/kv/kvserver/protectedts: delete ptcache

According to the TODOs and context on #118772, this cache has been broken
for years but wasn't fully removed as planned in 24.x.

Release note: none.
Epic: none.

dt committed:
protectedts: remove deprecatedSpans from MakeRecord and friends

Getting close to removing it entirely but there are still several references
to the proto field. But this mostly removes it from the PTS public API.

Release note: none.
Epic: none.

stevendanna commented on pkg/kv/kvserver/protectedts/ptstorage/storage.go:
```diff
@@ -76,28 +76,18 @@ func (p *storage) Protect(ctx context.Context, r *ptpb.Record) error {
 		len(r.DeprecatedSpans), encodedTarget, encodedTarget}
 
-	if updateMeta {
-		query = protectQuery
```
Note for other commit-by-commit reviewers: `protectQuery` and others are removed in subsequent commit.

stevendanna commented on pkg/kv/kvserver/protectedts/ptstorage/storage.go:
```diff
@@ -78,15 +70,23 @@ func (p *storage) Protect(ctx context.Context, r *ptpb.Record) error {
 	r.DeprecatedSpans = nil
 	encodedTarget, err := protoutil.Marshal(&ptpb.Target{Union: r.Target.GetUnion(),
 		IgnoreIfExcludedFromBackup: r.Target.IgnoreIfExcludedFromBackup})
+	if err != nil { // how can this possibly fail?
```
Existing code so no need to change, but I'm not sure this comment is useful in this part of the code. Might be better to just express the belief it can't fail by making the error an AssertionFailedf.

stevendanna commented on pkg/kv/kvserver/protectedts/ptstorage/storage.go:
```diff
@@ -78,15 +70,23 @@ func (p *storage) Protect(ctx context.Context, r *ptpb.Record) error {
+	s := makeSettings(p.settings)
+	query := protectQuery
 	args := []interface{}{
+		s.maxSpans, s.maxBytes, 0, // num_spans is always 0 now since we use targets
```
Not important, but I wonder if we could update the query to not have to pass this zero here.

stevendanna reviewed (approved):
🔥 Nice.

Looks like there aren't too many more references to the DeprecatedSpans field so perhaps a follow-up PR can remove it completely.  There is also probably some cleanup we can do to the SQL queries themselves. But, neither of those should get in the way of this.

dt commented:
I concur with all observations that we could clean up even more here, but I'm probably not going to spend more time here right away so leaning towards merging this now rather than letting perfect be the enemy of good, or at least better.

TFTR!

bors r+
